autogpt_platform/backend/backend/data/model.py

```diff
@@ -434,6 +434,15 @@ class OAuthState(BaseModel):
     code_verifier: Optional[str] = None
     """Unix timestamp (seconds) indicating when this OAuth state expires"""
     scopes: list[str]
+    # Fields for external API OAuth flows
+    callback_url: Optional[str] = None
+    """External app's callback URL for OAuth redirect"""
+    state_metadata: dict[str, Any] = Field(default_factory=dict)
+    """Metadata to echo back to external app on completion"""
+    api_key_id: Optional[str] = None
+    """ID of the API key that initiated this OAuth flow"""
+    is_external: bool = False
```

Comment:
this field can be derived directly from whether the `callback_url` is set so not needed